DFReader.py

```diff
@@ -605,8 +617,8 @@ class DFReader_text(DFReader):
     def __init__(self, filename, zero_time_base=False):
         DFReader.__init__(self)
         # read the whole file into memory for simplicity
-        f = open(filename, mode='r')
-        self.lines = f.readlines()
+        with open(filename, 'r') as f:
+            self.lines = f.readlines()
         f.close()
```

Comment:
@khancyr Actually, sorry, mavlogdump.py is working just fine on master for dumping text-based logs.

This is the exception under python2:

```
pbarker@bluebottle:~/rc/pymavlink((HEAD detached at khancyr/python_correction1))$ mavlogdump.py ~/crash2.log
Traceback (most recent call last):
  File "/home/pbarker/.local/bin/mavlogdump.py", line 4, in <module>
    __import__('pkg_resources').run_script('pymavlink==2.0.8', 'mavlogdump.py')
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 719, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1504, in run_script
    exec(code, namespace, namespace)
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/EGG-INFO/scripts/mavlogdump.py", line 58, in <module>
    zero_time_base=args.zero_time_base)
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/pymavlink/mavutil.py", line 1232, in mavlink_connection
    m = DFReader.DFReader_text(device, zero_time_base=zero_time_base)
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/pymavlink/DFReader.py", line 628, in __init__
    self.init_clock()
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/pymavlink/DFReader.py", line 377, in init_clock
    m = self.recv_msg()
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/pymavlink/DFReader.py", line 456, in recv_msg
    return self._parse_next()
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/pymavlink/DFReader.py", line 689, in _parse_next
    self._add_msg(m)
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/pymavlink/DFReader.py", line 483, in _add_msg
    self.flightmode = mavutil.mode_string_acm(m.Mode)
  File "/home/pbarker/.local/lib/python2.7/site-packages/pymavlink-2.0.8-py2.7-linux-x86_64.egg/pymavlink/mavutil.py", line 1683, in mode_string_acm
    return "Mode(%u)" % mode_number
TypeError: %u format: a number is required, not unicode
pbarker@bluebottle:~/rc/pymavlink((HEAD detached at khancyr/python_correction1))$ which python
```

And under python3:
```
pbarker@bluebottle:~/rc/pymavlink((HEAD detached at khancyr/python_correction1))$ python3 /home/pbarker/.local/bin/mavlogdump.py ~/crash2.log
Traceback (most recent call last):
  File "/home/pbarker/.local/bin/mavlogdump.py", line 4, in <module>
    __import__('pkg_resources').run_script('pymavlink==2.0.8', 'mavlogdump.py')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 719, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1504, in run_script
    exec(code, namespace, namespace)
  File "/home/pbarker/.local/lib/python3.5/site-packages/pymavlink-2.0.8-py3.5-linux-x86_64.egg/EGG-INFO/scripts/mavlogdump.py", line 55, in <module>
    zero_time_base=args.zero_time_base)
  File "/home/pbarker/.local/lib/python3.5/site-packages/pymavlink-2.0.8-py3.5-linux-x86_64.egg/pymavlink/mavutil.py", line 1227, in mavlink_connection
    from pymavlink import DFReader
  File "/home/pbarker/.local/lib/python3.5/site-packages/pymavlink-2.0.8-py3.5-linux-x86_64.egg/pymavlink/DFReader.py", line 35, in <module>
    "q": ("q", None, long),  # Backward compat
NameError: name 'long' is not defined
pbarker@bluebottle:~/rc/pymavlink((HEAD detached at khancyr/python_correction1))$ 
```